lab_5_scrapper/scrapper.py

```diff
@@ -120,46 +252,159 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Finds meta information of article
         """
-        pass
+        self.article.title = article_soup.find('h1', class_="article__title").text
+        date = article_soup.find(class_="article__meta-date")
+        self.article.date = self.unify_date_format(date.text)
+        self.article.topics = [topic.text for
+                               topic in article_soup.find_all('a', class_="article-list__tag")]
+        self.article.author = ["NOT FOUND"]
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        if not re.search(r'\d{4}', date_str):
+            curr_year = datetime.date.today().year
+            date_str = re.sub(r'(?<=[А-Яа-я])(?=,\s\d{2})', f' {curr_year}', date_str)
+
+        ru_eng_months = {
+            "янв": "jan",
+            "фев": "feb",
+            "мар": "mar",
+            "апр": "apr",
+            "мая": "may",
+            "июн": "jun",
+            "июл": "jul",
+            "авг": "aug",
+            "сен": "sep",
+            "окт": "oct",
+            "ноя": "nov",
+            "дек": "dec"
+        }
+
+        ru_month = re.search(r"[А-Яа-я]{3}", date_str).group()
+        date_str = date_str.replace(ru_month, ru_eng_months[ru_month])
+        return datetime.datetime.strptime(date_str, '%d %b  %Y, %H:%M')
 
     def parse(self) -> Union[Article, bool, list]:
         """
         Parses each article
         """
-        pass
+        page = make_request(self.full_url, self.config)
+        article_bs = BeautifulSoup(page.content, "lxml")
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
 
 
 def prepare_environment(base_path: Union[Path, str]) -> None:
     """
     Creates ASSETS_PATH folder if no created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(parents=True)
+
+
+class CrawlerRecursive(Crawler):
+    """
+    Recursive crawler implementation
+    """
+
+    def __init__(self, config: Config) -> None:
+        super().__init__(config)
+        self.start_url = config.get_seed_urls()[0]
+        self.all_urls = []
+        self._load_crawler_data()
+
+    def _load_crawler_data(self) -> None:
+        current_path = Path(__file__)
+        crawler_data_path = current_path.parent / 'crawler_data.json'
+        if crawler_data_path.exists():
+            with open('crawler_data.json', 'r', encoding='utf-8') as f:
+                crawler_data = json.load(f)
+            self.start_url, self.urls, self.all_urls = crawler_data.values()
+
+    def _save_crawler_data(self) -> None:
+        crawler_data = {
+            'start_url': self.start_url,
+            'urls': self.urls,
+            'all_urls': self.all_urls
+        }
+        with open('crawler_data.json', 'w', encoding='utf-8') as f:
```

Comment:
create all non-git assets in <project_root>/build or any other git ignored folder. do you know where to find them?

Reply:
Done